### ISSUE TITLE
Clickhouse max connections is always 1

### ISSUE THREAD

**Issue by caseyduquettesc** [getsentry/snuba#5775]
### Environment

Self-hosted 24.3.0

### Steps to Reproduce

Noticed an odd log with `CLICKHOUSE_MAX_CONNECTIONS` set to 25

> Connection pool is full, discarding connection: clickhouse-cs-production.default.svc.cluster.local. Connection pool size: 1

This makes me believe this value isn't used at all. Correct me if I'm wrong, but shouldn't `max_connections` be set to `cluster["max_connections"]` here?

https://github.com/getsentry/snuba/blob/e295b24de506f1ac8a98c61e000d41a388298855/snuba/clusters/cluster.py#L391-L409

**Comment by caseyduquettesc** [comment]
Possibly related - https://github.com/getsentry/snuba/issues/4596

**Comment by bkk-bcd** [comment]
Yeah this is a complete mess for us, total breakage for ingestion. Renders Sentry 💯 unusable 

See the monkey patching hilarity that's ensuing: https://github.com/sentry-kubernetes/charts/pull/1175

This is what seems to have broke it, and AFAICT nothing in the code actually takes the configuration into account. But then I'm not a python guy so: https://github.com/getsentry/snuba/pull/4988

**Comment by bkk-bcd** [comment]
> ### Environment
> Self-hosted 24.3.0
> 
> ### Steps to Reproduce
> Noticed an odd log with `CLICKHOUSE_MAX_CONNECTIONS` set to 25
> 
> > Connection pool is full, discarding connection: clickhouse-cs-production.default.svc.cluster.local. Connection pool size: 1
> 
> This makes me believe this value isn't used at all. Correct me if I'm wrong, but shouldn't `max_connections` be set to `cluster["max_connections"]` here?
> 
> https://github.com/getsentry/snuba/blob/e295b24de506f1ac8a98c61e000d41a388298855/snuba/clusters/cluster.py#L391-L409

Yeah that looks like exactly what was missed in https://github.com/getsentry/snuba/pull/4988
